library/alloc/src/rc.rs

```diff
@@ -933,6 +934,28 @@ impl<T: ?Sized> Rc<T> {
         Weak { ptr: this.ptr }
     }
 
+    /// Convert a reference to an [`Rc`] into a reference to a [`Weak`] of the same type.
+    ///
+    /// This is a type-only operation; it doesn't modify the inner reference counts.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(rc_as_weak)]
+    ///
+    /// use std::rc::{Rc, Weak};
+    ///
+    /// let five: &Rc<i32> = &Rc::new(5);
+    ///
+    /// let weak_five: &Weak<i32> = Rc::as_weak(five);
+    /// ```
+    #[inline]
+    #[unstable(feature = "rc_as_weak", issue = "100472")]
+    #[must_use]
+    pub const fn as_weak<'a>(this: &'a Self) -> &'a Weak<T> {
+        unsafe { mem::transmute::<&'a Rc<T>, &'a Weak<T>>(this) }
```

Comment:
Casting pointers (`&*(this as *const Self as *const Weak<T>)`) is generally preferred to `transmute()` in cases like that (see [`transmute()` docs](https://doc.rust-lang.org/std/mem/fn.transmute.html#alternatives)).

Reply:
Excellent. Done.